src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide(hidden) {
+    this.hidden = !hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Carnivore || obj.hidden) {
+      return;
+    }
+
+    obj.health -= 50;
+
+    Animal.alive = Animal.alive.filter(beast => beast.health !== 0);
```

Comment:
wrap this in if condition, so that you don't filter an array each time you bite someone
also, health can be less then 0